File: spring-data-requery/src/main/java/org/springframework/data/requery/mapping/RequeryPersistentEntity.java
package org.springframework.data.requery.mapping;

import org.springframework.context.ApplicationContextAware;
import org.springframework.data.mapping.PersistentEntity;

import java.util.Collection;

/**
 * org.springframework.data.requery.mapping.RequeryPersistentEntity
 *
 * @author debop
 * @since 18. 6. 7
 */
public interface RequeryPersistentEntity<T> extends PersistentEntity<T, RequeryPersistentProperty>, ApplicationContextAware {

    RequeryPersistentProperty getSingleIdProperty();

    Collection<RequeryPersistentProperty> getIdProperties();

    Collection<RequeryPersistentProperty> getIndexes();

    Collection<RequeryPersistentProperty> getEmbeddedProperties();
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/mapping/RequeryPersistentEntity.kt
package org.springframework.data.requery.kotlin.mapping

import org.springframework.context.ApplicationContextAware
import org.springframework.data.mapping.PersistentEntity

/**
 * org.springframework.data.requery.mapping.RequeryPersistentEntity
 *
 * @author debop
 */
interface RequeryPersistentEntity<E>: PersistentEntity<E, RequeryPersistentProperty>, ApplicationContextAware {

    val singleIdProperty: RequeryPersistentProperty?

    val idProperties: Collection<RequeryPersistentProperty>

    val indexes: Collection<RequeryPersistentProperty>

    val embeddedProperties: Collection<RequeryPersistentProperty>

}